WordPress/wporg-internal-notes
# Add a note type for logging post events, that displays inline with user-input internal notes

Introduces a second type of note for logging changes to a post. These log notes appear in the sidebar along with user-created internal notes, and can provide extra context to what is happening with a post.

On the backend, this plugin provides logging for some generic event types that happen when posts are updated, but the idea is that other plugins (e.g. the Pattern Directory) can add their own logging as well.

In the UI, log notes have a more compact style in the notes list so that they are de-emphasized compared to user-created internal notes. They also can't be deleted.

~I have made an attempt to have new log notes appear automatically in the sidebar when changes to a post are saved, but right now, it's causing a problem because the new log is getting loaded twice, even though the note is only getting created once. It must be some kind of race condition with the way React component updates happen, but so far I haven't been able to figure out how to avoid it.~ Fixed thanks to @StevenDufresne 

## Screencast
https://d.pr/v/iVeFjS

## To test

1. You might need to rebuild your local dev environment. The `0-sandbox.php` file has been updated to add post support for the logging. Check out the updated readme for more info about how enabling logging works.
2. Create a new post, then switch to the Notes sidebar. You'll see some initial events get logged.
3. Try changing the post title and/or content.
4. Try changing the post's taxonomy terms.
5. Try adding a featured image, and changing the post's template.
6. Log notes should not be deletable.
7. Add an Internal Note using the "Add Note" form in the sidebar. Once you have both internal notes and log notes, try using the filter button to show only one or the other.
